algosdk/atomic_transaction_composer.py

```diff
@@ -703,3 +747,31 @@ def __init__(
         self.confirmed_round = confirmed_round
         self.tx_ids = tx_ids
         self.abi_results = results
+
+
+class DryrunABIResult:
```

Comment:
How about we inherit this from the `ABIResult` class? This constructor can call `super()` and set the cost.